core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1126,101 +1335,19 @@ private void processInitialMessage(long address, int remaining) throws PeerDisco
-
-        short parameterCount = getShort(lo);
-
-        if (parameterCount > 0) {
-            if (lo + Short.BYTES + parameterCount * Integer.BYTES > msgLimit) {
-                LOG.error()
-                        .$("could not read parameters [parameterCount=").$(parameterCount)
-                        .$(", offset=").$(lo - address)
-                        .$(", remaining=").$(msgLimit - lo)
-                        .$(']').$();
-                throw BadProtocolException.INSTANCE;
-            }
-
-            LOG.debug().$("params [count=").$(parameterCount).$(']').$();
-            lo += Short.BYTES;
-
-            bindVariableService.clear();
-            setupBindVariables(lo, parameterCount, bindVariableSetters);
-        } else if (parameterCount < 0) {
-            LOG.error()
-                    .$("invalid parameter count [parameterCount=").$(parameterCount)
-                    .$(", offset=").$(lo - address)
-                    .$(']').$();
-            throw BadProtocolException.INSTANCE;
-        }
-
-        // at this point we may have a current query that is not null
-        // this is ok to lose reference to this query because we have cache
-        // of all of them, which is looked up by query text
-
-        responseAsciiSink.reset();
-        final Object statement = factoryCache.peek(queryText);
-        if (statement == null) {
-            final CompiledQuery cc = compiler.compile(queryText, sqlExecutionContext);
-            sqlExecutionContext.storeTelemetry(cc.getType(), TelemetryOrigin.PG_WIRE);
-
-            switch (cc.getType()) {
-                case CompiledQuery.SELECT:
-                    currentFactory = cc.getRecordCursorFactory();
-                    queryTag = TAG_SELECT;
-                    factoryCache.put(queryText, currentFactory);
-                    break;
-                case CompiledQuery.INSERT:
-                    currentInsertStatement = cc.getInsertStatement();
-                    queryTag = TAG_INSERT;
-                    factoryCache.put(queryText, currentInsertStatement);
-                    break;
-                case CompiledQuery.COPY_LOCAL:
-                    queryTag = TAG_COPY;
-                    sendCopyInResponse(compiler.getEngine(), cc.getTextLoader());
-                    break;
-                default:
-                    // DDL SQL
-                    queryTag = TAG_OK;
-                    prepareParseComplete();
-                    prepareReadyForQuery(responseAsciiSink);
-                    LOG.info().$("executed DDL").$();
-                    send();
-                    break;
-            }
-        } else {
-            if (statement instanceof RecordCursorFactory) {
+    private void setupNamedStatement(ObjList<BindVariableSetter> bindVariableSetters, CharSequenceObjHashMap<NamedStatementWrapper> namedStatementMap, CharSequence statementName) throws SqlException {
+        if (statementName != null) {
```

Comment:
for performance reasons this null check should be done outside of the method call. The rationale is that method call does not happen when statement is null